contracts/contracts/strategies/ThreePoolStrategy.sol

```diff
@@ -110,6 +107,48 @@ contract ThreePoolStrategy is InitializableAbstractStrategy {
         );
     }
 
+    /**
+     * @dev Deposit the entire balance of any supported asset into the Curve 3pool
+     */
+    function depositAll() external onlyVault nonReentrant {
+        uint256[3] memory _amounts = [uint256(0), uint256(0), uint256(0)];
+        uint256 depositValue = 0;
+        ICurvePool curvePool = ICurvePool(platformAddress);
+
+        for (uint256 i = 0; i < assetsMapped.length; i++) {
+            uint256 balance = IERC20(assetsMapped[i]).balanceOf(address(this));
```

Comment:
assetsMapped[i] is referenced 4 times in the code below. My understanding is that lookups are pretty expensive in solidity.
Does the solidity compiler automatically optimize this and create temporary memory storage for these lookups or does it execute the lookup instructions 4 times? If the latter it would be worth creating a temporary variable to store assetsMapped[i] to minimize gas costs.

Reply:
Good question, I created a temporary variable. Would be great to have an understanding of what optimisations the compiler might make here (if any).